qcodes/dataset/experiment_container.py

```diff
@@ -209,9 +226,11 @@ def new_experiment(name: str,
         the new experiment
     """
     conn = conn or connect(get_DB_location())
-    return Experiment(name=name, sample_name=sample_name,
-                      format_string=format_string,
-                      conn=conn)
+    experiment = Experiment(
+        name=name, sample_name=sample_name, format_string=format_string, conn=conn
+    )
+    _set_active_experiment_id(experiment.exp_id)
```

Comment:
We could also move this into the Experiment constructor

Reply:
@astafan8 what do you think?

Reply:
I initially wanted to do that, but then decided to put it into individual functions.
This move has a benefit that if someone directly use Experiment class, then this active_experiment also get triggered.

Reply:
my perspective is that the global side effect, which is setting the active experiment, should not be attached to the object itself so that there exists a way to create an experiment without activating it immediately. hence I agree with @FarBo .

so i think let's keep setting active experiment outside of the experiment object